SIPS/sip-29.md

```diff
@@ -0,0 +1,232 @@
+
+## Specification
+
+> Indented sections like this are considered non-normative.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definitions
+
+1. In this document, all definitions are written in TypeScript.
+
+2. Any time an asset needs to be identified, it MUST use the [CAIP-19][caip-19]
+   representation.
+
+### Snap Manifest
+
+This SIP introduces a new permission named `endowment:assets`.
+This permission grants a Snap the ability to provide asset information to the client.
+
+This permission is specified as follows in `snap.manifest.json` files:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:assets": {
+      "scopes": [
+        "bip122:000000000019d6689c085ae165831e93"
+      ]
+    }
+  }
+}
+```
+
+`scopes` - A non-empty array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+
+### Snap Implementation
+
+Two methods are defined in the Snap Assets API:
+
+Any Snap that wishes to provide asset information **MUST** implement the following API:
+
+#### Get Asset Metadata
+
+```typescript
+import { OnAssetLookupHandler } from "@metamask/snaps-sdk";
+
+export const onAssetLookup: OnAssetLookupHandler = async ({
+  assets
+}) => {
+  const assetsMetadata = /* Get metadata */;
+  return { assets: assetsMetadata };
+};
+```
+
+The type for an `onAssetLookup` handler function’s arguments is:
+
+```typescript
+interface OnAssetLookupArgs {
+  assets: Caip19AssetType[];
+}
+```
+
+The type for an `onAssetLookup` handler function’s return value is:
+
+```typescript
+type OnAssetLookupReturn = {
+  assets: Record<Caip19AssetType, AssetMetadata>;
+};
+```
+
+#### Get Asset Conversion Rate
+
+```typescript
+import { OnAssetConversionHandler } from "@metamask/snaps-sdk";
+
+export const onAssetConversion: OnAssetConversionHandler = async ({
+  conversions
+}) => {
+  const conversionRates = /* Get conversion rate */;
+  return { conversionRates };
+};
+```
+
+The type for an `onAssetConversion` handler function’s arguments is:
+
+```typescript
+type Conversion = {
+  from: Caip19AssetType;
+  to: Caip19AssetType;
+};
+
+type OnAssetConversionArgs = {
+  conversions: Conversion[];
+};
+```
+
+The type for an `onAssetConversion` handler function’s return value is:
+
+```typescript
+type AssetConversionRate = {
+  // The rate of conversion from the source asset to the target asset. It
+  // means that 1 unit of the `from` asset should be converted to this amount
+  // of the `to` asset.
+  rate: string;
+
+  // The UNIX timestamp of when the conversion rate was last updated.
+  conversionTime: number;
+
+  // The UNIX timestamp of when the conversion rate will expire.
+  expirationTime: number;
```

Comment:
hey @danroc 
The data we currently have does not include MarketData, which means we cannot display the 1-day price changes or chart-based price changes. I believe this should be updated to include an interface like the following:
```
interface MarketData {
	id: string;
	price: number;
	marketCap: number;
	allTimeHigh: number;
	allTimeLow: number;
	totalVolume: number;
	high1d: number;
	circulatingSupply: number;
	dilutedmarketCap: number;
	marketCapPercentChange1d: number;
	priceChange1d: number;
	pricePercentChange1h: number;
	pricePercentChange1d: number;
	pricePercentChange7d: number;
	pricePercentChange14d: number;
	pricePercentChange30d: number;
	pricePercentChange200d: number;
	pricePercentChange1y: number;
}
```

Reply:
Hey @salimtb, thanks! We didn’t plan to include market data in the first release, so we left it out of this API to give ourselves more time to figure out what’s really needed.

Are there any fields you’d like to add or remove from the current EVM implementation? Now might be a good time to make any changes you’ve been thinking about.

Also, it’s not a big deal if it gets added later to this SIP or even a new one.

Reply:
thanks for your response, @danroc. Not all the fields are currently in use, but we decided to keep them for potential future features. If you'd like, I can provide you with a list of the properties that are currently being used.
all of the properties are returned by the endpoint `/spot-price` of the price API